src/ucp/core/ucp_worker.c

```diff
@@ -1201,13 +1201,19 @@ static void ucp_worker_close_ifaces(ucp_worker_h worker)
     UCS_ASYNC_UNBLOCK(&worker->async);
 }
 
-static ucs_sys_device_t
-ucp_worker_iface_get_sys_device(const ucp_worker_iface_t *wiface)
+static const ucp_tl_resource_desc_t *
+ucp_worker_iface_get_tl_resource(const ucp_worker_iface_t *wiface)
 {
     const ucp_context_h context     = wiface->worker->context;
     const ucp_rsc_index_t rsc_index = wiface->rsc_index;
 
-    return context->tl_rscs[rsc_index].tl_rsc.sys_device;
+    return &context->tl_rscs[rsc_index];
```

Comment:
can remove rsc_index variable, and maybe also context (if it fits to one line after the change)